language-design.md

```diff
@@ -60,27 +220,86 @@ What does the post on grayscale tell us about the process of API design?
 **Response**
 
 
+In Verou's post, the *process* of API design itself is extremely important and
+must consider the user's intuition and ability to deduce functionality from
+function names. It should be an interactive process that involves discussions
+and polls of potential users. For example, when determining if the command
+should be `gray`, `white`, `black`, or `rgb`, it might be advantageous to have
+prospective users vote and survey different communities for input. Names in APIs
+are difficult to create well and could have significant consequences in the
+understandability of the API which extends to its adoption by developers.
+
+Anthony could personally imagine debating for hours over whether to choose
+`black`, `white`, or `rgb` due to the different tradeoffs. While designing an
+API, often there are multiple decisions that might hold benefits over others.
+For example, being consistent with the base language is valuable yet there might
+be compatiability issues as discussed in the post.
+
+Overall, I think the post really offers insight on how there are many options to
+choose from when desiging an API: sometimes you either should pick an option and
+deal with the consequences but we think gathering information from users is
+essential as long as you are generally willing to consider it. Sometimes it is
+impractical to please every user but as Bloch says "aim to displease everyone
+equally. Most APIs are overconstrained."
```

Comment:
Documentation and tooling is the best way to handle things like this.  

As for a programmatic solution, I think this could be better designed to take a `Level` object, and then have well-named methods of creating or initializing a level.

I think it makes sense to do that, but at the same time plenty of languages handle it just fine.